### PR TITLE
Replace generic LinkedHashMap with Dictionary where applicable

### PR DESCRIPTION
**Pull request by bahusoid**
The only place where it seems matters is [`BatchFetchQueue`](https://github.com/nhibernate/nhibernate-core/blob/cc6baef36425e5f4a72689e3281560955835d68a/src/NHibernate/Engine/BatchFetchQueue.cs#L34) but not 100% sure.

### CODE REVIEW

File: src/NHibernate/Mapping/Table.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public enum SchemaAction
 	public class Table : IRelationalModel
 	{
 		private readonly List<string> checkConstraints = new List<string>();
-		private readonly LinkedHashMap<string, Column> columns = new LinkedHashMap<string, Column>();
+		private readonly Dictionary<string, Column> columns = new Dictionary<string, Column>();
```
Comment (hazzik) [inline src/NHibernate/Mapping/Table.cs:30]: I think it does make sense to have a linked hash map here: it ensures the order of elements. And the application does depend on the order of columns.

File: src/NHibernate/SqlCommand/SqlInsertBuilder.cs [hunk 2]
```diff
@@ -19,7 +20,7 @@ public class SqlInsertBuilder : ISqlStringBuilder
 		private string comment;
 
 		// columns-> (ColumnName, Value) or (ColumnName, SqlType) for parametrized column
-		private readonly LinkedHashMap<string, object> columns = new LinkedHashMap<string, object>();
+		private readonly Dictionary<string, object> columns = new Dictionary<string, object>();
```
Comment (hazzik) [inline src/NHibernate/SqlCommand/SqlInsertBuilder.cs:23]: Again, order of columns is important.

File: src/NHibernate/SqlCommand/SqlUpdateBuilder.cs [hunk 3]
```diff
@@ -19,7 +19,7 @@ public class SqlUpdateBuilder : SqlBaseBuilder, ISqlStringBuilder
 		private string comment;
 
 		// columns-> (ColumnName, Value) or (ColumnName, SqlType) for parametrized column
-		private readonly LinkedHashMap<string, object> columns = new LinkedHashMap<string, object>();
+		private readonly Dictionary<string, object> columns = new Dictionary<string, object>();
```
Comment (hazzik) [inline src/NHibernate/SqlCommand/SqlUpdateBuilder.cs:22]: Order of columns is important here.

**Comment by hazzik** [comment]
Order of columns in a table definitely does matter. In the insert/update sql builders it might matter, depending on how the parameters get bound to the command.

**Comment by bahusoid** [comment]
Yeah. But in Dictionary order is the same too as long elements from the dictionary are not deleted. It's a known behavior too so I see no problem here

**Comment by hazzik** [comment]
It is confirmed, but not guaranteed. I feel like it's better to be safe than sorry.

**Comment by bahusoid** [comment]
> but not guaranteed

I don't believe it would ever change and it's better to avoid using some custom baked classes. But Ok my belief is not very strong argument :)

**Comment by fredericDelaporte** [comment]
I am agree it is better to not rely on this long known behavior of dictionary, even though I have already relied on it in some circumstances in other projects. We already have the code for avoiding relying on it in NHibernate, better keep it that way.
We can not be sure all implementations of the standard will always have this behavior.

**Pull request closed without merging**